packages/react/src/hooks/useStorageURL.ts

```diff
@@ -14,31 +14,35 @@ export interface UseStorageURLResult {
  */
 export const useStorageURL = (
   key: string,
-  options?: S3ProviderGetConfig
+  options?: S3ProviderGetConfig,
+  fallbackURL?: string
 ): UseStorageURLResult & { fetch: () => () => void } => {
   const [result, setResult] = React.useState<UseStorageURLResult>({
     isLoading: true,
   });
 
   // Used to prevent an infinite loop on useEffect, because `options`
   // will have a different reference on every render
-  const serializedOptions = JSON.stringify(options);
+  const optionsRef = React.useRef(options);
```

Comment:
Just for my own understanding why is options different on each render?

Reply:
If a user call `useStorageURL` and construct the option in line, then a new object will be created on each render and thus causes an infinite loop in `useEffect`

```jsx
useStorageURL(
    imgKey,
    { xxx }
  )
```